From 3acc9c435940d8343882252e04f0d6c97e225ee7 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@armbian.com>
Date: Wed, 26 Jun 2024 08:25:25 -0400
Subject: [PATCH] Add k1-bananapi-f3 dts

K1-X deb1 board seems to have only one PCIe lane wired for PCIe C port,
as a mini PCIe slot.

Fix the PCIe lane number, otherwise the PCIe controller could not be
correctly brought up and no card will be detected.
Signed-off-by: Icenowy Zheng <uwu@icenowy.me>

Bring up bluetooth via DTS
Signed-off-by: Patrick Yavitz <pyavitz@armbian.com>
---
 arch/riscv/boot/dts/spacemit/Makefile         |  1 +
 .../boot/dts/spacemit/k1-bananapi-f3.dts      | 26 +++++++++++++++++++
 2 files changed, 27 insertions(+)
 create mode 100644 arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts

diff --git a/arch/riscv/boot/dts/spacemit/Makefile b/arch/riscv/boot/dts/spacemit/Makefile
index 2be95dbca797..d198630de062 100644
--- a/arch/riscv/boot/dts/spacemit/Makefile
+++ b/arch/riscv/boot/dts/spacemit/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 
 dtb-$(CONFIG_SOC_SPACEMIT_K1X) += k1-x_deb1.dtb
+dtb-$(CONFIG_SOC_SPACEMIT_K1X) += k1-bananapi-f3.dtb
 
 obj-$(CONFIG_BUILTIN_DTB) += $(addsuffix .o, $(dtb-y))
diff --git a/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts b/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts
new file mode 100644
index 000000000000..cdb9c5a77668
--- /dev/null
+++ b/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/* Copyright (c) 2024 Spacemit, Inc */
+
+/dts-v1/;
+
+#include "k1-x_deb1.dts"
+
+/ {
+	model = "BananaPi BPI-F3";
+};
+
+&pcie2_rc {
+	num-lanes = <1>;
+};
+
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "okay";
+
+	bluetooth {
+		compatible = "realtek,rtl8852bs-bt";
+		device-wake-gpios = <&gpio 64 GPIO_ACTIVE_HIGH>;
+		host-wake-gpios = <&gpio 65 GPIO_ACTIVE_HIGH>;
+	};
+};
-- 
2.39.2

